atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -90,24 +90,63 @@ public static SerializableTransactionManager create(
             Schema schema,
             Environment env,
             boolean allowHiddenTableAccess) {
-        return create(config, ImmutableSet.of(schema), env, allowHiddenTableAccess);
+        return create(config, Optional.absent(), ImmutableSet.of(schema), env, allowHiddenTableAccess);
+    }
+
+    /**
+     * Create a {@link SerializableTransactionManager} with provided configuration,
+     * {@link SSLSocketFactory}, {@link Schema}, and an environment in which to
+     * register HTTP server endpoints.
+     *
+     * This method is deprecated in favor of specifying the ssl configuration for
+     * client-side leader connections via {@link LeaderConfig}'s sslConfiguration
+     * and not through the sslSocketFactory argument.
```

Comment:
still not clear on why it's preferred this thing know about SslConfiguration, IMO this is the preferred method: user tells you which SSLSocketFactory to use rather than requiring you to get one from a random trustStore.